extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/MSQTestTaskActionClient.java

```diff
@@ -40,18 +41,29 @@
 import org.apache.druid.timeline.SegmentId;
 import org.joda.time.Interval;
 
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
 
 public class MSQTestTaskActionClient implements TaskActionClient
 {
 
-  private static final String VERSION = "test";
+  public static final String VERSION = "test";
   private final ObjectMapper mapper;
   private final ConcurrentHashMap<SegmentId, AtomicInteger> segmentIdPartitionIdMap = new ConcurrentHashMap<>();
+  private final Map<String, List<Interval>> usedIntervals = ImmutableMap.of(
+      "foo", ImmutableList.of(Intervals.of("2001-01-01/2001-01-04"), Intervals.of("2000-01-01/2000-01-04")),
+      "foo2", ImmutableList.of(Intervals.of("2000-01-01/P1D"))
+  );
+  private final Set<DataSegment> publishedSegments = new HashSet<>();
 
-  public MSQTestTaskActionClient(ObjectMapper mapper)
+  public MSQTestTaskActionClient(
```

Comment:
Yes, CalciteTests for MSQ only work on testing the SELECT engine of MSQ (since the original Calcite tests do not have any analog for ingestion). Changing this doesn't affect the test cases for MSQ.